src/CachePlugin.php

```diff
@@ -34,31 +35,86 @@
     private $config;
 
     /**
+     * Cache directives indicating if a response can not be cached.
+     *
+     * @var array
+     */
+    private $noCacheFlags = ['no-cache', 'private'];
+
+    /**
      * @param CacheItemPoolInterface $pool
      * @param StreamFactory          $streamFactory
      * @param array                  $config        {
      *
-     *     @var bool $respect_cache_headers Whether to look at the cache directives or ignore them
-     *     @var int $default_ttl (seconds) If we do not respect cache headers or can't calculate a good ttl, use this
-     *              value
-     *     @var string $hash_algo The hashing algorithm to use when generating cache keys
-     *     @var int $cache_lifetime (seconds) To support serving a previous stale response when the server answers 304
-     *              we have to store the cache for a longer time than the server originally says it is valid for.
-     *              We store a cache item for $cache_lifetime + max age of the response.
-     *     @var array $methods list of request methods which can be cached.
-     * }
+     * @var bool   $respect_cache_headers Whether to look at the cache directives or ignore them. This option is
+     *             deprecated, use `respect_response_cache_directives` instead
+     * @var int    $default_ttl (seconds) If we do not respect cache headers or can't calculate a good ttl, use this
+     *             value
+     * @var string $hash_algo The hashing algorithm to use when generating cache keys
+     * @var int    $cache_lifetime (seconds) To support serving a previous stale response when the server answers 304
+     *             we have to store the cache for a longer time than the server originally says it is valid for.
+     *             We store a cache item for $cache_lifetime + max age of the response.
+     * @var array  $methods list of request methods which can be cached
+     * @var array  $respect_response_cache_directives list of cache directives this plugin will respect while caching responses.
```

Comment:
Yeah I think this was fix for styleCi but indeed I've reverted the indent changes!